docs: add curl --resolve fallback for nip.io DNS issues in quick-start-controlplane (#742)
C5C3/forge: docs/quick-start-controlplane.md

curl -k https://keystone.127-0-0-1.nip.io:8443/v3
```

::: tip If your host cannot resolve `*.nip.io`

> Some local resolvers (observed on macOS) fail to resolve `nip.io` hostnames,

Reviewer: Normally this is not related to MacOS. Do you use a PiHole or some other kind of DNS filter in your local setup?

Reviewer 2: It seems that my router is actively filtering or blocking this specific DNS pattern.

Reviewer: Then adjust the note accordingly.

Reviewer 2: done

curl -k https://keystone.127-0-0-1.nip.io:8443/v3
```

::: tip If your host cannot resolve `*.nip.io`
On some local setups that filter or block this specific DNS pattern, the
following command fails:

```bash
curl -k https://keystone.127-0-0-1.nip.io:8443/v3
```

The command returns the following error:

```
curl: (6) Could not resolve host: keystone.127-0-0-1.nip.io
```

`nip.io` deliberately resolves hostnames like `keystone.127-0-0-1.nip.io` to
the loopback address `127.0.0.1`. Some routers ship **DNS rebind
protection**, a security feature that silently drops any DNS response
resolving a public hostname to a private or loopback address, which is

> exactly this pattern. When your machine's resolver is such a router, `nip.io`

Reviewer 3: ```suggestion
this pattern. When your machine's resolver does this, `nip.io`
```

> never resolves, regardless of the OS. Confirm the cause by comparing your

Reviewer 3: ```suggestion
never resolves. Confirm the cause by comparing your
```

router against a public resolver:

```bash
dig +short keystone.127-0-0-1.nip.io @<your-router-ip>   # empty: blocked
dig +short keystone.127-0-0-1.nip.io @1.1.1.1            # 127.0.0.1: as expected
```

> The most robust fix is host-local and independent of the router: add loopback

Reviewer 3: ```suggestion
The most reliable fix is host-local: add loopback
```

entries to `/etc/hosts` (covers `curl`, the `openstack` CLI, and a browser for

> the Horizon dashboard alike):

Reviewer 3: ```suggestion
entries to `/etc/hosts`:
```

```bash
sudo sh -c 'cat >> /etc/hosts <<EOF
127.0.0.1 keystone.127-0-0-1.nip.io
127.0.0.1 horizon.127-0-0-1.nip.io

> 127.0.0.1 glance.127-0-0-1.nip.io

Reviewer 3: Does hosts allow regexp or glob?

Reviewer 2: No. /etc/hosts does not support regular expressions, globs, or wildcard entries. Each hostname must be listed explicitly:
